compiler/rustc_const_eval/src/interpret/call.rs

```diff
@@ -752,11 +753,41 @@ impl<'tcx, M: Machine<'tcx>> InterpCx<'tcx, M> {
         &mut self,
         fn_val: FnVal<'tcx, M::ExtraFnVal>,
         (caller_abi, caller_fn_abi): (ExternAbi, &FnAbi<'tcx, Ty<'tcx>>),
-        args: &[FnArg<'tcx, M::Provenance>],
+        mut args: Vec<FnArg<'tcx, M::Provenance>>,
         with_caller_location: bool,
     ) -> InterpResult<'tcx> {
         trace!("init_fn_tail_call: {:#?}", fn_val);
 
+        let mut local_temps = vec![];
+        let frame_locals = &self.stack().last().unwrap().locals;
+        if frame_locals.iter().any(|frame_local| frame_local.is_allocation()) {
+            // Allocations corresponding to the locals in the last frame.
+            let local_allocs: FxHashSet<_> = frame_locals
+                .iter()
+                .filter_map(|local| local.as_mplace_or_imm()?.left()?.0.provenance?.get_alloc_id())
+                .collect();
```

Comment:
Can you wrap this in `std::cell::LazyCell`, so that we don't create a hash map if none of the arguments have mplaces?